interface/patient_file/encounter/find_code_dynamic.php

```diff
@@ -226,76 +226,83 @@ function SelectList(jobj) {
+                    echo "   </select></div>\n";
+                }
+                echo "&nbsp;&nbsp;\n";
+                echo "<div class='col'>";
+                echo "<input type='checkbox' name='form_include_inactive' value='1' onclick='oTable.fnDraw()' />" .
+                xlt('Include Inactive') . "\n";
+                echo "&nbsp;&nbsp;\n";
+                echo "<button class='btn btn-secondary btn-sm btn-delete' value='" . xla('Delete') . "' onclick='delcode()'>" . xla('Delete') . "</button>\n";
+                echo "<select name='form_delcodes'>\n";
+                echo " <option value=''>" . xlt('All') . "</option>\n";
+                echo "</select>\n";
+                echo "&nbsp;&nbsp;\n";
+                echo "<button class='btn btn-secondary btn-sm btn-cancel' value='" . xla('Close') . "' onclick='dlgclose()'>" . xla('Close') . "</button>\n";
+                echo "</div>";
             }
-            echo "</select>\n";
-        }
-    } else {
-        echo "<select name='form_code_type' onchange='oTable.fnDraw()'>\n";
-        foreach ($code_types as $key => $value) {
-            echo " <option value='" . attr($key) . "'";
-            echo ">" . xlt($value['label']) . "</option>\n";
-        }
-        echo " <option value='PROD'";
-        echo ">" . xlt("Product") . "</option>\n";
-        echo "   </select>\n";
-    }
-    echo "&nbsp;&nbsp;\n";
-    echo "<input type='checkbox' name='form_include_inactive' value='1' onclick='oTable.fnDraw()' />" .
-    xlt('Include Inactive') . "\n";
-    echo "&nbsp;&nbsp;\n";
-    echo "<input type='button' value='" . xla('Delete') . "' onclick='delcode()' />\n";
-    echo "<select name='form_delcodes'>\n";
-    echo " <option value=''>" . xlt('All') . "</option>\n";
-    echo "</select>\n";
-    echo "&nbsp;&nbsp;\n";
-    echo "<input type='button' value='" . xla('Close') . "' onclick='dlgclose()' />\n";
-}
-if ($what == 'lists') {
-    echo "<input type='button' value='" . xla('Delete') . "' onclick='SelectList({\"code\":\"\"})' />\n";
-}
-echo "</p>\n";
-?>
-
-<table id="my_data_table" class="table table-striped table-hover table-sm" style="width: 90%;">
- <thead>
-  <tr>
-   <th><?php echo xlt('Code'); ?></th>
-   <th><?php echo xlt('Description'); ?></th>
-  </tr>
- </thead>
- <tbody>
-  <tr>
-   <!-- Class "dataTables_empty" is defined in jquery.dataTables.css -->
-   <td colspan="2" class="dataTables_empty">...</td>
-  </tr>
- </tbody>
-</table>
-
-<?php if ($what == 'fields' && $source == 'E') { ?>
-<center>
-<p>
-<input type='text' name='new_field_id' size='20' />&nbsp;
-<input type='button' value='<?php echo xla('Or create this new field ID') ?>' onclick='newField()' />
-</p>
-</center>
-<?php } ?>
-
-</form>
+            if ($what == 'lists') {
+                echo "<button class='btn btn-secondary btn-sm btn-delete' value='" . xla('Delete') . "' onclick='SelectList({\"code\":\"\"})'>" . xla('Delete') . "\n";
```

Comment:
missing `</button>`

Reply:
also please don't get in the habit of translating button value attr unless value is going to be the buttons ui name. One or the other. Say a button is going to be a submit button and you translate spanish value attr which may be used in the submit conditional value which is in english then they won't match. 
I see this often enough to point it out again.

Reply:
This can be a bit murky since depends on the mechanism of the input. At this point, @stu01509 is simply keeping the value there as is in order to not break whatever the mechanism was. For example, submit buttons need to keep the value (generally just needs to be anything, but you don't know for sure unless look at the script that catches the $_REQUEST). There was time when a significant part of OpenEMR actually used the value when catching variables, which broke horribly on translations. In those cases, the name attribute was then used instead, so the translated value was not used (ie. value could be anything). Since it's not straightforward and case specific and the goal is super high throughput, it seems ok to just keep the value's as is, which will also avoid bugs since it does not change behavior. Agree that we definitely should not carry over this process in new scripts.

Reply:
I'm to tired to debate the point. It's enough to say that it's a bad habit.

Reply:
I also agree it's a really bad habit, and will ban the original author (whomever and wherever you are) from OpenEMR for 1 minute.

Reply:
ahh, I feel so much better now!

Reply:
The ban is over (whomever and wherever you are). If it makes you feel better, I probably left  very large dents in the walls with my head wherever I lived during the translation effort that got to deal with translated values.

Reply:
I get it. I think you guys did a great job on translations. I don't know how it could be any easier to use.

Reply:
Thanks for the @bradymiller sir explanation, At first I also removed the button value attribute, but never think of it will break submit form data. If we will refactor it may a big plan and need a lot of tests.

Reply:
@sjpadgett @bradymiller The value attribute of button elements (even though in some cases breaks code) is a sin and should be burned at the steak. The value of buttons is what resides inside the buttons making the value attribute useless.

Reply:
Yes, agreed in theory. But it will break code in many cases, not just some.